### PR TITLE
Drop special support for rst2pdf

### PR DESCRIPTION
**Pull request by tk0miya**
For some historical reason, we give a special treatment to rst2pdf.
But it is not maintained for long time, and it does not support python3.
So I think it's time to drop these special treatment. And I also dropped the mentions for rst2pdf on our docs.

This does not mean rst2pdf does not work with Sphinx.

### CODE REVIEW

File: sphinx/registry.py [hunk 1]
```diff
@@ -332,8 +332,6 @@ def load_extension(self, app, extname):
 
         if metadata is None:
             metadata = {}
-            if extname == 'rst2pdf.pdfbuilder':
-                metadata['parallel_read_safe'] = True
```
Comment (tk0miya) [inline sphinx/registry.py:336]: With this removal, `rst2pdf` will be warned "the rst2pdf extension does not declare if it is safe for parallel reading" if users use `-j` option.

**Review by stephenfin — APPROVED** [review]
Yup, totally agree.

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Thank you for reviewing! I'll merge this after sending a PR to rst2pdf project tomorrow.

**Comment by tk0miya** [comment]
Just posted a PR: https://github.com/rst2pdf/rst2pdf/pull/640

**Pull request merged**